controllers/machinehealthcheck_controller.go

```diff
@@ -259,9 +262,57 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 		if annotations.IsPaused(cluster, t.Machine) {
 			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
 		} else {
-			logger.Info("Target has failed health check, marking for remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
-			conditions.MarkFalse(t.Machine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediationReason, clusterv1.ConditionSeverityWarning, "MachineHealthCheck failed")
+			if m.Spec.RemediationTemplate != nil {
+				// If external remediation request already exists,
+				// return early
+				if r.externalRemediationRequestExists(ctx, m, t.Machine.Name) {
+					return ctrl.Result{}, nil
+				}
+
+				cloneOwnerRef := &metav1.OwnerReference{
+					APIVersion: clusterv1.GroupVersion.String(),
+					Kind:       "Machine",
+					Name:       t.Machine.Name,
+					UID:        t.Machine.UID,
+				}
+
+				from, err := external.Get(ctx, r.Client, m.Spec.RemediationTemplate, t.Machine.Namespace)
+				if err != nil {
+					conditions.MarkFalse(m, clusterv1.ExternalRemediationTemplateAvailable, clusterv1.ExternalRemediationTemplateNotFound, clusterv1.ConditionSeverityError, "External Remediation Template not found")
+					return ctrl.Result{}, errors.Wrapf(err, "error retrieving remediation template %v %q for machine %q in namespace %q within cluster %q", m.Spec.RemediationTemplate.GroupVersionKind(), m.Spec.RemediationTemplate.Name, t.Machine.Name, t.Machine.Namespace, m.Spec.ClusterName)
+				}
+
+				generateTemplateInput := &external.GenerateTemplateInput{
+					Template:    from,
+					TemplateRef: m.Spec.RemediationTemplate,
+					Namespace:   t.Machine.Namespace,
+					ClusterName: t.Machine.ClusterName,
+					OwnerRef:    cloneOwnerRef,
+				}
+				to, err := external.GenerateTemplate(generateTemplateInput)
+				if err != nil {
+					return ctrl.Result{}, errors.Wrapf(err, "failed to create template for remediation request %v %q for machine %q in namespace %q within cluster %q", m.Spec.RemediationTemplate.GroupVersionKind(), m.Spec.RemediationTemplate.Name, t.Machine.Name, t.Machine.Namespace, m.Spec.ClusterName)
```

Comment:
Either here or in a follow-up PR: let's mark the condition `clusterv1.ExternalRemediationTemplateAvailable` false here, so the user is aware. Looking at `GenerateTemplate`, the only way it returns an error is if there is a problem getting `spec.template` from the template CR.

Reply:
I'm +1 to reconsider this in another PR.

Reply:
Let's reconsider in another PR.